src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/4.diff.sql

```diff
@@ -447,6 +477,101 @@ AS
     COMMIT TRANSACTION
 GO
 
+/***************************************************************************************/
+-- STORED PROCEDURE
+--     GetExtendedQueryTagErrors
+--
+-- DESCRIPTION
+--     Gets the extended query tag errors by tag path.
+--
+-- PARAMETERS
+--     @tagPath
+--         * The TagPath for the extended query tag for which we retrieve error(s).
+--
+-- RETURN VALUE
+--     The tag error fields and the corresponding instance UIDs.
+/***************************************************************************************/
+CREATE OR ALTER PROCEDURE dbo.GetExtendedQueryTagErrors (@tagPath VARCHAR(64))
+AS
+BEGIN
+    SET NOCOUNT     ON
+    SET XACT_ABORT  ON
+
+    DECLARE @tagKey INT
+    SELECT @tagKey = TagKey
+    FROM dbo.ExtendedQueryTag WITH(HOLDLOCK)
+    WHERE dbo.ExtendedQueryTag.TagPath = @tagPath
+
+    -- Check existence
+    IF (@@ROWCOUNT = 0)
+        THROW 50404, 'extended query tag not found', 1 
+
+    SELECT
+        TagKey,
+        ErrorMessage,
+        CreatedTime,
+        StudyInstanceUid,
+        SeriesInstanceUid,
+        SopInstanceUid
+    FROM dbo.ExtendedQueryTagError AS XQTE
+    INNER JOIN dbo.Instance AS I
+    ON XQTE.Watermark = I.Watermark
+    WHERE XQTE.TagKey = @tagKey
+END
+GO
+
+/***************************************************************************************/
+-- STORED PROCEDURE
+--     AddExtendedQueryTagError
+--
+-- DESCRIPTION
+--    Adds an Extended Query Tag Error or Updates it if exists.
+--
+-- PARAMETERS
+--     @tagKey
+--         * The related extended query tag's key
+--     @errorMessage
+--         * The error message
+--     @watermark
+--         * The watermark
+--
+-- RETURN VALUE
+--     The tag key of the error added.
+/***************************************************************************************/
+CREATE OR ALTER PROCEDURE dbo.AddExtendedQueryTagError (
+    @tagKey INT,
+    @errorMessage VARCHAR(128),
+    @watermark BIGINT
+)
+AS
+    SET NOCOUNT     ON
+    SET XACT_ABORT  ON
+    BEGIN TRANSACTION
+
+    DECLARE @currentDate DATETIME2(7) = SYSUTCDATETIME()
```

Comment:
Do we want to use time provided by code ? The error time could not be the time when writing to database.